lib/vquic/ngtcp2.c

```diff
@@ -833,15 +881,23 @@ static int cb_h3_recv_header(nghttp3_conn *conn, int64_t stream_id,
      !memcmp(":status", h3name.base, h3name.len)) {
     int status = decode_status_code(h3val.base, h3val.len);
     DEBUGASSERT(status != -1);
-    msnprintf(stream->mem, stream->len, "HTTP/3 %03d \r\n", status);
+    /* make room for the status line */
+    result = h3_buffer_grow(data, stream, 13);
+    if(result) {
+      return -1;
+    }
+    ncopy = msnprintf(stream->mem, stream->len, "HTTP/3 %03d \r\n", status);
   }
   else {
     /* store as a HTTP1-style header */
-    msnprintf(stream->mem, stream->len, "%.*s: %.*s\n",
-              h3name.len, h3name.base, h3val.len, h3val.base);
+    result = h3_buffer_grow(data, stream, h3name.len + 2 + h3val.len + 1);
+    if(result) {
+      return -1;
+    }
+    ncopy = msnprintf(stream->mem, stream->len, "%.*s: %.*s\n",
+                      h3name.len, h3name.base, h3val.len, h3val.base);
   }
 
-  ncopy = strlen(stream->mem);
```

Comment:
Seemed unnecessary to call `strlen` when `msnprintf` already returns the number of characters copied. `msnprintf` can return `-1` in case of error, but the only possible error condition is a bad format string, and here we use format strings that we know are valid.